### PR TITLE
fix: MoveTo() when only changing case.

### LINKED ISSUE #737: Discrepancy of MoveTo() when only changing case.
**Describe the bug**

When only changing case, (ex: abc.txt -> ABC.txt) `System.IO.Abstractions.TestingHelpers.MockFileInfo.MoveTo()` throw an IOException but `System.IO.Abstractions.FileInfoWrapper.MoveTo()` and `System.IO.FileInfo.MoveTo()` do not throw.

**To Reproduce**

```csharp
//System.IO.FileInfo.MoveTo()
string targetFilePath = @"D:\TestZone\abc.txt";
string renamedFilePath = @"D:\TestZone\ABC.txt";
File.WriteAllText(targetFilePath, "abc");
var dotNetFile = new FileInfo(targetFilePath);
dotNetFile.MoveTo(renamedFilePath);
File.Delete(renamedFilePath);

//System.IO.Abstractions.FileInfoWrapper.MoveTo()
var realFileSystem = new FileSystem();
realFileSystem.File.WriteAllText(targetFilePath, "abc");
var realFile = realFileSystem.FileInfo.FromFileName(targetFilePath);
realFile.MoveTo(renamedFilePath);
realFileSystem.File.Delete(renamedFilePath);

//System.IO.Abstractions.TestingHelpers.MockFileInfo.MoveTo()
var mockFileSystem = new MockFileSystem(new Dictionary<string, MockFileData>()
{
    [targetFilePath] = new MockFileData("abc"),
});
var mockFile = mockFileSystem.FileInfo.FromFileName(targetFilePath);
mockFile.MoveTo(renamedFilePath);
```

The last line throw an `IOException(The file D:\TestZone\ABC.txt already exists.)`.

**Expected behavior**
Not throw Exception.

**Additional context**

This is an aside, but `System.IO.DirectoryInfo.MoveTo()` throw `IOException(Source and destination path must be different)` in same case.

### PR DESCRIPTION
**Pull request by soi013**
fix #737

### CODE REVIEW

File: src/System.IO.Abstractions.TestingHelpers/MockFileInfo.cs [hunk 1]
```diff
@@ -230,6 +231,13 @@ public override FileSecurity GetAccessControl(AccessControlSections includeSecti
 
         public override void MoveTo(string destFileName)
         {
+            if (destFileName != FullName && Path.GetDirectoryName(destFileName) == DirectoryName
+                && destFileName.ToLowerInvariant() == FullName.ToLowerInvariant())
+            {
+                var tempPath = GetSafeTempName(destFileName);
+                MoveTo(tempPath);
+            }
```
Comment (fgreinacher) [inline src/System.IO.Abstractions.TestingHelpers/MockFileInfo.cs:239]: While this technically works, it looks a bit fishy to me. I'd guess there is some place deeper down the stack where things go wrong. Do you want to dive in @soi013 or should I take over?

Comment (soi013) [reply src/System.IO.Abstractions.TestingHelpers/MockFileInfo.cs:239]: ええ、私もスマートなコードだとは思わない。

より良い方法があったら、あなたにお任せしたい。

Comment (fgreinacher) [reply src/System.IO.Abstractions.TestingHelpers/MockFileInfo.cs:239]: Sure, see https://github.com/System-IO-Abstractions/System.IO.Abstractions/pull/740 ;)

**Comment by fgreinacher** [comment]
Closing in favor of https://github.com/System-IO-Abstractions/System.IO.Abstractions/pull/740

**Pull request closed without merging**